src/borg/archiver.py

```diff
@@ -107,6 +108,8 @@ def wrapper(self, args, **kwargs):
             with repository:
                 if manifest or cache:
                     kwargs['manifest'], kwargs['key'] = Manifest.load(repository)
+                    if args.__dict__.get('compression'):
```

Comment:
that looks strange, esp. considering that you do an attribute access in the next line.